packages/core/src/helpers/calls.ts

```diff
@@ -20,29 +20,37 @@ export function warnOnInvalidCall(call: Call | Falsy) {
 /**
  * @internal Intended for internal use - use it on your own risk
  */
-export function encodeCallData(call: Call | Falsy, chainId: number, queryParams: QueryParams = {}): RawCall | Falsy {
+export function validateCall(call: Call | Falsy) {
   if (!call) {
-    return undefined
+    throw new Error('Call is falsy')
   }
   const { contract, method, args } = call
   if (!contract.address || !method) {
-    warnOnInvalidCall(call)
-    return undefined
+    throw new Error('Missing contract address or method name')
   }
+
   try {
-    const isStatic = queryParams.isStatic ?? queryParams.refresh === 'never'
-    const refreshPerBlocks = typeof queryParams.refresh === 'number' ? queryParams.refresh : undefined
+    contract.interface.encodeFunctionData(method, args)
+    return call
+  } catch (err: any) {
+    throw new Error(`Invalid contract call for method="${method}" on contract="${contract.address}": ${err.message}`)
```

Comment:
Try to include contract name in there, if it exists

Reply:
I think it's not possible to extract name from Contract instance